cli/src/cast.rs

```diff
@@ -195,8 +195,11 @@ async fn main() -> eyre::Result<()> {
                 config.eth_rpc_url.unwrap_or_else(|| "http://localhost:8545".to_string()),
             )?;
 
+            let chain_id = provider.get_chainid().await?;
+            let chain = Chain::try_from(chain_id.as_u64())?;
+
             let mut builder =
-                TxBuilder::new(&provider, config.sender, address, eth.chain, false).await?;
+                TxBuilder::new(&provider, config.sender, address, chain, false).await?;
```

Comment:
Maybe instead of failing here we should turn it into an `Option` and fall back on `eth.chain`?

Reply:
Fall back in case of what? All providers must implement `eth_chainId` so that shouldn't fail. Or are you thinking that `Chain::try_from` could fail?

Reply:
Yeah, I think `Chain::try_from` could fail

Reply:
Yeah I think your right. Someone might be using a chain that isn't yet in the mapping, and we wouldn't want that to brick it. Just pushed a change.